code/game/objects/structures/crates_lockers/closets.dm

```diff
@@ -144,6 +144,9 @@
 	var/atom/L = drop_location()
 	for(var/atom/movable/AM in src)
 		AM.forceMove(L)
+		for(var/i in src)
+			var/atom/movable/thing = i
+			SEND_SIGNAL(thing, COMSIG_CLOSET_CONTENTS_EXPOSED, src)
```

Comment:
I don't like that we're calling this on every single item inside the closet. I think this should fire on the closet itself, `decomposition` should use `connect_loc`, then if the location is a locker, it registers to it.